File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/auth_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';

class CrawChatAuthModule {
  final CrawChatSdkContext context;

  CrawChatAuthModule(this.context);

  Future<PortalLoginResponse?> login(PortalLoginRequest body) async {
    final session = await context.backendClient.auth.login(body);
    final accessToken = session?.accessToken;
    if (accessToken != null && accessToken.isNotEmpty) {
      useToken(accessToken);
    }
    return session;
  }

  Future<PortalMeResponse?> me() {
    return context.backendClient.auth.me();
  }

  void useToken(String token) {
    context.setAuthToken(token);
  }

  void clearToken() {
    context.clearAuthToken();
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/src/auth-module.ts
import type {
  CrawChatAuthLoginRequest,
  CrawChatAuthLoginResult,
  CrawChatAuthSession,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatAuthModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  async login(body: CrawChatAuthLoginRequest): Promise<CrawChatAuthLoginResult> {
    const session = await this.context.backendClient.auth.login(body);
    if (session.accessToken) {
      this.useToken(session.accessToken);
    }
    return session;
  }

  me(): Promise<CrawChatAuthSession> {
    return this.context.backendClient.auth.me();
  }

  useToken(token: string): void {
    this.context.setAuthToken(token);
  }

  clearToken(): void {
    this.context.clearAuthToken();
  }
}
